sql/core/src/main/scala/org/apache/spark/sql/execution/joins/HashedRelation.scala

```diff
@@ -459,8 +459,8 @@ private[execution] final class LongToUnsafeRowMap(val mm: TaskMemoryManager, cap
    */
   def getValue(key: Long, resultRow: UnsafeRow): UnsafeRow = {
     if (isDense) {
-      val idx = (key - minKey).toInt
-      if (idx >= 0 && key <= maxKey && array(idx) > 0) {
+      val idx = (key - minKey).toInt  // could overflow
+      if (key >= minKey && key <= maxKey && array(idx) > 0) {
```

Comment:
Ah yeah OK this should be OK. I might suggest the following as a little simpler, but whatever:

```
if (key >= minKey && key <= maxKey) {
  val value = array((key - minKey).toInt)
  if (value > 0) {
    return getRow(value, resultRow)
  }
}
```
?